pkgs/development/node-packages/default.nix

```
@@ -199,6 +199,25 @@ let
       '';
     };
 
+    manta = super.manta.override {
+      nativeBuildInputs = [ pkgs.nodejs-12_x pkgs.nodePackages.json ];
+      postInstall = ''
+        # create completions, following upstream procedure https://github.com/joyent/node-manta/blob/v5.2.3/Makefile#L85-L91
+        mkdir -p $out/share/bash-completion/completions
+
+        completion_cmds=$(find ./bin -type f)
+        completion_file=$out/share/bash-completion/completions/manta
+
+        node ./lib/create_client.js
+        echo "# node-manta tools v$(cat package.json | json version) completion" >> $completion_file
+        echo $completion_cmds | xargs -n1 basename | sed -E 's/(.*)/#   \1(1)/' >> $completion_file
+        echo "" >> $completion_file
+        for cmd in $completion_cmds; do \
+          $cmd --completion | grep -v '^#' >> $completion_file; \
+        done
```

```suggestion
        completion_cmds=$(find ./bin -type f)

        node ./lib/create_client.js
        for cmd in $completion_cmds; do
          installShellCompletion --cmd $cmd --bash <($cmd --completion)
        done
```
the path already contains the version. Not sure what the other thing is but I don't think we need that.

---

at the moment, the head of the final completion file looks like this:
```
# node-manta tools v5.2.3 completion
#   mput(1)
#   mln(1)
#   msign(1)
#   mrm(1)
#   muntar(1)
#   mjob(1)
#   mls(1)
#   mrmdir(1)
#   mchmod(1)
#   mget(1)
#   minfo(1)
#   mlogin(1)
#   mmkdir(1)
#   mmpu(1)
#   mchattr(1)
#   mmd5(1)
#   mfind(1)
```
I agree that we don't need the version string, I just adapted everything as it was from upstream.

The other comments are all the commands that the file provides completion for, don't know if that's a nice to have or we just want to leave it out, too?

---

Answering this myself: after a quick test, it's really unneeded, because we generate a `$cmd.bash` file for each command instead of one big completion file holding all the subcommands. It's a lot better like this, because before the resulting file had to be `source`d in an extra step afterwards.